docs/Migrations.rst

```diff
@@ -0,0 +1,260 @@
+Migrations
+==================
+
+Neo4j.rb provides an ``ActiveRecord``-like migration framework and a set of helper methods to manipulate both database schema and data.
+
+
+Generators
+------------------
+
+Migrations can be created by using the built-in Rails generator:
+
+.. code-block:: bash
+
+  rails generate neo4j:migration RenameUserNameToFirstName
+
+This will generate a new file located in ``db/neo4j/migrate/xxxxxxxxxx_rename_user_name_to_first_name.rb``
+
+.. code-block:: ruby
+  class RenameUserNameToFirstName < Neo4j::Migrations::Base
+    def up
+    end
+
+    def down
+    end
+  end
+
+In the same way as ``ActiveRecord`` does, you should fill up the ``up`` and ``down`` methods to define the migration and (eventually) the rollback steps.
+
+
+Transactions
+------------------
+Every migrations runs inside a transaction by default. So, if some statement fails inside a migration fails, the database rollbacks to the previous state.
+
+However this behaviour is not always good. For instance, neo4j doesn't allow schema and data changes in the same transaction.
+
+To disable this, you can use the ``disable_transactions!`` helper in your migration definition:
+
+.. code-block:: ruby
+
+  class SomeMigration < Neo4j::Migrations::Base
+    disable_transactions!
+
+    ...
+  end
+
+Neo4j::Migrations::Helpers
```

Comment:
Perhaps a title like "Migration Helpers" so that people don't think they need to know about the `Neo4j::Migrations::Helpers" class.

Also, perhaps the `RenameUserNameToFirstName` migration example should use at least one helper so that people can make the association about where they would call the methods below.